src/ir_Daikin.cpp

```diff
@@ -261,25 +268,457 @@ void IRDaikinESP::setSwingHorizontal(bool state) {
+    clearBit(DAIKIN_BYTE_EYE, DAIKIN_BIT_EYE);
+}
+
+bool IRDaikinESP::getEye() {
+  return (getBit(DAIKIN_BYTE_EYE, DAIKIN_BIT_EYE) > 0);
+}
+
+
+void IRDaikinESP::setMold(bool state) {
+  if (state)
+    setBit(DAIKIN_BYTE_MOLD, DAIKIN_BIT_MOLD);
+  else
+    clearBit(DAIKIN_BYTE_MOLD, DAIKIN_BIT_MOLD);
+}
+
+bool IRDaikinESP::getMold() {
+  return (getBit(DAIKIN_BYTE_MOLD, DAIKIN_BIT_MOLD) > 0);
+}
+
+
+void IRDaikinESP::setBit(uint8_t byte, uint8_t bitmask) {
+  daikin[byte] |= bitmask;
+  checksum();
+}
+
+
+void IRDaikinESP::clearBit(uint8_t byte, uint8_t bitmask) {
+  bitmask = ~bitmask;
+  daikin[byte] &= bitmask;
+  checksum();
+}
+
+uint8_t IRDaikinESP::getBit(uint8_t byte, uint8_t bitmask) {
+  uint8_t ret;
+  ret = daikin[byte] & bitmask;
+  return ret;
+}
+
+// starttime: Number of minutes after midnight, in 10 minutes increments
+void IRDaikinESP::enableOnTimer(uint16_t starttime) {
+  daikin[13] |= 0b00000010;
+  uint16_t lopbits;
+  lopbits = starttime;
+  lopbits &= 0x00FF;
+  daikin[18] = lopbits;
+  starttime = starttime >> 8;
+  // only keep 4 bits
+  daikin[19] &= 0xF0;
+  daikin[19] |= starttime;
+  checksum();
+}
+
+void IRDaikinESP::disableOnTimer() {
+  enableOnTimer(0x600);
+  daikin[13] &= 0b11111101;
+  checksum();
+}
+
+uint16_t IRDaikinESP::getOnTime() {
+  uint16_t ret;
+  ret = daikin[19] & 0x0F;
+  ret = ret << 8;
+  ret += daikin[18];
+  return ret;
+}
+bool IRDaikinESP::getOnTimerEnabled() {
+    return daikin[13] & 0b00000010;
+}
+
+// endtime: Number of minutes after midnight, in 10 minutes increments
+void IRDaikinESP::enableOffTimer(uint16_t endtime) {
+  daikin[13] |= 0b00000100;
+  uint16_t lopbits;
+  lopbits = endtime;
+  lopbits &= 0x0FF0;
+  lopbits = lopbits >> 4;
+  daikin[20] = lopbits;
+  endtime &= 0x000F;
+  endtime = endtime << 4;
+  daikin[19] &= 0x0F;
+  daikin[19] |= endtime;
+  checksum();
+}
+
+void IRDaikinESP::disableOffTimer() {
+  enableOffTimer(0x600);
+  daikin[13] &= 0b11111011;
+  checksum();
+}
+
+uint16_t IRDaikinESP::getOffTime() {
+  uint16_t ret, tmp;
+  ret = daikin[20];
+  ret = ret << 4;
+  tmp = daikin[19] & 0xF0;
+  tmp = tmp >> 4;
+  ret += tmp;
+  return ret;
+}
+
+bool IRDaikinESP::getOffTimerEnabled() {
+  return daikin[13] & 0b00000100;
+}
+
+void IRDaikinESP::setCurrentTime(uint16_t time) {
+  uint16_t lopbits;
+  lopbits = time;
+  lopbits &= 0x00FF;
+  daikin[5] = lopbits;
+  time = time >> 8;
+  // only keep 4 bits
+  daikin[6] &= 0xF0;
+  daikin[6] |= time;
+  checksum();
+}
+
+uint16_t IRDaikinESP::getCurrentTime() {
+  uint16_t ret;
+  ret = daikin[6] & 0x0F;
+  ret = ret << 8;
+  ret += daikin[5];
+  return ret;
+}
+
+String IRDaikinESP::renderTime(uint16_t timemins) {
+  uint16_t hours, mins;
+  hours = timemins / 60;
+  mins = timemins - (hours * 60);
+  String ret = String(mins);
+  if (ret.length() == 1)
+    ret = "0" + ret;
+  ret = ":" + ret;
+  ret = String(hours) + ret;
+  return ret;
+}
+
+
+#if DAIKIN_DEBUG
+
+void IRDaikinESP::printState() {
```

Comment:
Ditto the comment on the previous method.
`Serial` & `String` are Arduino-only/specific.